userops_ext_test.go

```diff
@@ -1892,3 +1893,1269 @@ func TestDataCopy_SetCrossChainIntent(t *testing.T) {
+				require.Equal(t, embedOp.InitCode, extractedOp.InitCode)
+				require.Equal(t, embedOp.CallData, extractedOp.CallData)
+			}
+		})
+	}
+}
+
+// Helper function to create a cross-chain intent for testing
+func createCrossChainIntent(t *testing.T, fromChain, toChain int64) *pb.Intent {
+	return &pb.Intent{
+		From: &pb.Intent_FromAsset{
+			FromAsset: &pb.Asset{
+				Address: "0x0A7199a96fdf0252E09F76545c1eF2be3692F46b",
+				Amount:  &pb.BigInt{Value: big.NewInt(100).Bytes()},
+				ChainId: &pb.BigInt{Value: big.NewInt(fromChain).Bytes()},
+			},
+		},
+		To: &pb.Intent_ToAsset{
+			ToAsset: &pb.Asset{
+				Address: "0x6B5f6558CB8B3C8Fec2DA0B1edA9b9d5C064ca47",
+				Amount:  &pb.BigInt{Value: big.NewInt(50).Bytes()},
+				ChainId: &pb.BigInt{Value: big.NewInt(toChain).Bytes()},
+			},
+		},
+	}
+}
+
+// TestUserOperation_Aggregate_DifferentOperations verifies behavior when aggregating different operations
+func TestUserOperation_Aggregate_DifferentOperations(t *testing.T) {
+	baseOp := mockCreateOp()
+	intent1 := createCrossChainIntent(t, 1, 56)
+	intentJSON1, err := protojson.Marshal(intent1)
+	require.NoError(t, err)
+
+	baseOp.CallData = intentJSON1
+
+	data1, err := baseOp.EncodeCrossChainCallData(EntrypointV06, mockOtherOpHash, true, intentJSON1)
+	require.NoError(t, err)
+
+	baseOp.CallData = data1
+	baseOp.Signature = mockSignature()
+
+	// Create first embedded op
+	embedOp1 := mockCreateOp()
+	intent2 := createCrossChainIntent(t, 56, 1)
+	intentJSON2, err := protojson.Marshal(intent2)
+	require.NoError(t, err)
+
+	embedOp1.CallData = intentJSON2
+
+	data2, err := embedOp1.EncodeCrossChainCallData(EntrypointV06, mockOtherOpHash, true, intentJSON2)
+	require.NoError(t, err)
+
+	embedOp1.CallData = data2
+	embedOp1.Signature = mockSignature()
+
+	// Aggregate first op
+	err = baseOp.Aggregate(embedOp1)
+	require.NoError(t, err)
+
+	// Store signature after first aggregation
+	firstAggregateSignature := make([]byte, len(baseOp.Signature))
+	copy(firstAggregateSignature, baseOp.Signature)
+
+	// Create second embedded op with different nonce
+	embedOp2 := mockCreateOp()
+	embedOp2.Nonce = big.NewInt(999)
+	embedOp2.CallData = data2
+	embedOp2.Signature = mockSignature()
+
+	// Aggregate second op
+	err = baseOp.Aggregate(embedOp2)
+	require.NoError(t, err)
+
+	// Verify signature changed
+	require.False(t, bytes.Equal(firstAggregateSignature, baseOp.Signature),
+		"Signature should change when aggregating different operation")
+
+	// Verify new operation is valid
+	status, err := baseOp.Validate()
+	require.NoError(t, err)
+	require.Equal(t, UnsolvedAggregateUserOp, status)
+
+	// Extract and verify the new embedded operation matches embedOp2
+	extractedOp, err := baseOp.ExtractEmbeddedOp()
+	require.NoError(t, err)
+	require.Equal(t, embedOp2.Nonce.String(), extractedOp.Nonce.String())
+}
+
+func TestUserOperation_Aggregate_ErrorHandling(t *testing.T) {
```

Comment:
A test aggregating an aggregate userOp? 
Any result is acceptable. A clear error message or double aggregation/extraction.